src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngineUnitTest.java

```diff
@@ -0,0 +1,200 @@
+                { Arrays.asList(SNP_C_100), "CAAAAAAAAA", "1X9M" },
+                { Arrays.asList(SNP_C_109), "AAAAAAAAAC", "9M1X" },
+                { Arrays.asList(SNP_C_105, SNP_C_106), "AAAAACCAAA", "5M2X3M" },
+
+                { Arrays.asList(DEL_AA_105), "AAAAAAAAA", "6M1D3M" },
+                { Arrays.asList(DEL_AA_100), "AAAAAAAAA", "1M1D8M" },
+                { Arrays.asList(DEL_AA_105, SNP_C_109), "AAAAAAAAC", "6M1D2M1X" },
+                { Arrays.asList(DEL_AA_105, SNP_C_107, SNP_C_109), "AAAAAACAC", "6M1D1X1M1X" },
+
+                { Arrays.asList(INS_TT_105),  "AAAAAATAAAA", "6M1I4M" },
+                { Arrays.asList(INS_GGG_106), "AAAAAAAGGAAA", "7M2I3M" },
+                { Arrays.asList(DEL_AA_100, INS_GGG_106, SNP_C_109), "AAAAAAGGAAC", "1M1D5M2I2M1X" },
+
+                //this tests that SNPS can be inserted immediately prior to (and following) indels
+                { Arrays.asList( SNP_C_105, DEL_AA_105 ), "AAAAACAAA", "5M1X1D3M" },
+                { Arrays.asList( SNP_C_100, DEL_AA_100 ), "CAAAAAAAA", "1X1D8M" },
+                { Arrays.asList( SNP_C_100, DEL_AA_100, SNP_G_102 ), "CGAAAAAAA", "1X1D1X7M" },
+                { Arrays.asList( SNP_C_105, INS_TT_105 ), "AAAAACTAAAA", "5M1X1I4M" },
+                { Arrays.asList( SNP_C_100, INS_TT_100, SNP_G_101 ), "CTGAAAAAAAA", "1X1I1X8M" },
+                { Arrays.asList( SNP_C_100, INS_TT_100, SNP_G_101, SNP_C_105, DEL_AA_105 ), "CTGAAACAAA", "1X1I1X3M1X1D3M" },
+
+                //testing that the logic around anchor bases isn't resulting in variants being dropped accidntally
+                { Arrays.asList( SNP_C_104, DEL_AA_105 ), "AAAACAAAA", "4M1X1M1D3M" },
+                { Arrays.asList( SNP_C_104, INS_TT_105 ), "AAAACATAAAA", "4M1X1M1I4M" },
+
+        };
+    }
+    @Test(dataProvider = "testConstructHaplotypeFromVariantsDataProvider")
+    public void basicConstructHaplotypeFromVariants(List<VariantContext> variants, String expectedBases, String expectedCigar) {
+        Haplotype ref = new Haplotype("AAAAAAAAAA".getBytes(), true, 500, TextCigarCodec.decode("10M"));
+        ref.setGenomeLocation(new SimpleInterval("20", 100, 110));
+
+        Haplotype result = PartiallyDeterminedHaplotypeComputationEngine.constructHaplotypeFromVariants(ref, variants, true);
+        Assert.assertEquals(result.getBases(), expectedBases.getBytes());
+        Assert.assertEquals(result.getCigar(), TextCigarCodec.decode(expectedCigar));
+
+        // Assert that the resulting event map matches the input variants:
+        //TODO these need to be smarter about compounds
+        EventMap resultEMap = result.getEventMap();
+//        Assert.assertEquals(resultEMap.getNumberOfEvents(), variants.size());
+//        for (VariantContext v : variants) {
+//            VariantContext actualVC = resultEMap.get(v.getStart());
+//            Assert.assertNotNull(actualVC);
+//            Assert.assertEquals(actualVC.getAlleles(), v.getAlleles());
```

Comment:
So this test failed because the tests i wrote above have alternative representations that realignment was causing the event map to be mismatching on... I've rewritten it to at least somewhat account for this fact... not perfect but alright and better than not testing the event set at all